src/transport.js

```diff
@@ -53,11 +54,18 @@ module.exports = function (swarm) {
       // specific for the transport we are using
       const proxyConn = new Connection()
 
-      next(multiaddrs.shift())
+      each(multiaddrs, next)
```

Comment:
@jackkleeman - yep, that was a great catch overall, I've patched my local fork with the simpler sequential `next(shift)` so far. This might have been the culprit of a bunch of weird issues I was running into at some point with the the WS vs TCP order of dialing https://github.com/libp2p/js-libp2p-circuit/pull/1#issuecomment-286959112 🥇